presto-spi/src/main/java/com/facebook/presto/spi/ConnectorPageSource.java

```diff
@@ -62,4 +65,9 @@
     @Override
     void close()
             throws IOException;
+
+    default CompletableFuture<?> isBlocked()
```

Comment:
@martint, @electrum and I talked this over and we agreed that what you have is the best API, so no changes are need to this.